src/lib.rs

```diff
@@ -19,6 +19,23 @@ fn normalize(license_string: &str) -> String {
     list.join(" OR ")
 }
 
+fn get_proc_macro_node_names(metadata: &Metadata, opt: &GetDependenciesOpt) -> HashSet<String> {
```

Comment:
This function name doesn't shows the result value is affected by `opt.avoid_proc_macros`.
How about removing `if opt.avoid_proc_macros` and adding it at the calling site of this function?

Reply:
I could, but I think it'll make things less clear. The pattern is being used elsewhere already. If we moved everything out to the calling site, it sort of defeats the purpose of the functions.
https://github.com/onur/cargo-license/blob/33b857a6b565ba9b5eb6a996d64b7671ed0033d5/src/lib.rs#L22